### ISSUE TITLE
Log file doesn't get cleared if xlwings isn't found

### ISSUE THREAD

**Issue by aglucas** [ZoomerAnalytics/xlwings#477]
I am using xlwings 0.7.2 on a mac with Excel 2016, and python3

Python 3 has been a pain to run on mac because mac has python2.7 preinstalled and is difficult to remove.  That said let me get to the issue.

I have a script `db_update.py` that has
`#!/usr/bin/env python` 
at the top I get an error about not finding  `from appscript import app, mactypes`
This happens because python 2.7 was run and it does not have xlwings.
Ok so I fix this with 
`#!/usr/bin/env python3`
But the error does not go away.
So I close everything Excel and Eclipse which I am running in debug mode to test the python code.
Now run a very simple script named `test1.py` that gets data from cell A1 multiples by 2 and puts result in A2.
This fails with the same error as before and reports the script causing the error is called `db_update.py`
But I ran the script `test1.py`
So I uninstall Excel, reboot the mac, reinstall Excel, boot again.
Now run `test1.py` and no problems are reported.
Now I run a script `calc_all.py` which has the same problem as `db_update.py` where it referenced python not python.
Boom same error as before but now the file causing the error is `calc_all.py`

Now run `test1.py` it fails and reports the failing script is `test1.py`
Back to rebuilding.
Uninstall Excel, reboot, install Excel, reboot.
Run Excel with `test1.py` and no error.

What is going on? 
How can I fix it.
It looks like the system

 is caching the errors and not clearing them out.

I am not sure if it is a bug in Excel or xlwings.

**Comment by fzumstein** [comment]
My recommendation is to use the Anaconda/Miniconda distribution on Mac, this usually gets rid of such problems. Amongst all moving parts, it sounds like Eclipse is the most likely reason to mess up.

Test your scripts in a Terminal by calling the proper python installation with your script name, e,g. `/usr/bin/python test1.py`. Make sure that the correct interpreter is started by executing `/usr/bin/python` (or whatever your path is).

**Comment by aglucas** [comment]
I just did a little more work on this and realized what actually causes the error.
It is when I run from excel using VB to call RunPython 
In this case Eclipse is not involved at all
RunPython "from calc_interactive import xlwings_live; xlwings_live()"
Once I get an error from this the only way to remove is to uninstall Excel

**Comment by aglucas** [comment]
Maybe I can get around this by modifying ExecuteMac to use a different pythonpath.
But this still does not fix the stuck error.

**Comment by fzumstein** [comment]
I seriously doubt that you need to uninstall Excel ;)

The way it works is that by default it takes your python installation that is invoked if you call `python` from a Terminal. Check out which it is by writing `which python` at a Terminal.

If that does not start the python interpreter you would like, you can either specify the full path to the interpreter under the VBA settings (`PYTHON_MAC`), see here: http://docs.xlwings.org/en/stable/vba.html#settings

or you can you run the following commands one after another in a Terminal by providing the correct path to the `bin` folder of your python installation:
```
echo export PATH="/path/to/your/python/bin:$PATH" >> ~/.bash_profile
source .bash_profile
```

**Comment by aglucas** [comment]
I did the bash_profile change already.
The annoying part is I can not get the error in excel to go away.
Seriously once I get an error from python its stuck in excel.  Even if I run different script I will get an error reported from the last error even if the script I just ran has no errors.
The only way to remove an error is to uninstall.

**Comment by fzumstein** [comment]
Ok, there must be something going on then with the error log file...

The locations are indicated here: http://docs.xlwings.org/en/stable/vba.html#log-file-default-locations

For some reason it doesn't get deleted - so for the moment removing it manually from `~/Library/Containers/com.microsoft.Excel/Data/xlwings_log.txt` should help and save you from a reinstallation.

**Comment by aglucas** [comment]
Yea that does it removing the xlwings_log.txt clears the old errors.
I repeated the process twice just to make sure.
At least now I don't have to uninstall and reinstall.
Thanks for looking into this.

Thanks
Aaron

**Comment by fzumstein** [comment]
So now the question is why it doens't remove it. Can you double check that your VBA xlwings module is also v0.7.2?

**Comment by aglucas** [comment]
Hello Felix
Thanks for your help today.
The quick response was great.
I am running the VBA xlwings 0.7.2  I checked that early on.

I also realized it might be tricky to recreate.
This is specifically what I had to do.
Use the default settings for python in xlwings 0.7.2
Install python3 on my machine
Open a spreadsheet the has buttons used to call VBA which calls runpython
At this point Excel seems hung and reports Running at the bottom left of
the spreadsheet.

Now is the funny part, what do I do next?
I chose to run from Eclipse the same script that is called from runpython
This script runs and I see the correct cells in Excel change.
But when it is done I get the described error.

This might be a very random sequence, and it might be worth checking what
happens when Excel is closed when it is "running"

Thanks
Aaron

**Comment by fzumstein** [comment]
Python needs to terminate (either successfully or with an Exception) in order for Excel to reset the 'Running' indication correctly in the status bar and to clear the error log. what was the error you got in the log that wasn't shown?

**Comment by aglucas** [comment]
If I run with the default PythonMac path I get the following.

 I close and reopen excel when it hangs "running" run python

I get the following error
"
Traceback (most recent call last):

  File "<string>", line 1, in <module>

  File "/Users/alucas/modeling/arch/delphi/db_update.py", line 3, in
<module>

    from xlwings import Workbook

  File
"/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/site-packages/xlwings/__init__.py",
line 24, in <module>

    from . import _xlmac as xlplatform

  File
"/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/site-packages/xlwings/_xlmac.py",
line 5, in <module>

    from appscript import app, mactypes

ImportError: No module named app script

New I run a different script called test.py from Eclipse which interacts
with the active worksheet.

"Traceback (most recent call last):

  File "<string>", line 1, in <module>

  File "/Users/alucas/modeling/arch/delphi/db_update.py", line 3, in
<module>

    from xlwings import Workbook

  File
"/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/site-packages/xlwings/__init__.py",
line 24, in <module>

    from . import _xlmac as xlplatform

  File
"/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/site-packages/xlwings/_xlmac.py",
line 5, in <module>

    from appscript import app, mactypes

ImportError: No module named app script"

Next rm  ~/Library/Containers/com.microsoft.Excel/Data/xlwings_log.txt

Now run
test1.py and no error

**Comment by fzumstein** [comment]
Ah right, yeah, so you basically run into the *Note* issue as mentioned here: http://docs.xlwings.org/en/stable/debugging.html

I'll see if the log file can be properly cleaned in that situation but not sure now.

**Comment by aglucas** [comment]
Yea that's it

**Comment by fzumstein** [comment]
log deleting with Mac Excel 2016 fixed by: d9ccf5c80a817089bd2598f61c3eb971c969dda8
